### ISSUE TITLE
[XSpeeds] Missing Results for Some Categories

### ISSUE THREAD

**Issue by SCUR0** [Jackett/Jackett#15700]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
2024-11-09 16:05:21.1959 Info Starting Jackett v0.22.898
2024-11-09 16:05:21.5930 Info Environment version: 8.0.10 (/app/Jackett/)
2024-11-09 16:05:21.5988 Info OS version: Unix 6.6.56.0 (64bit OS) (64bit process)
2024-11-09 16:05:21.6009 Info Jackett variant: CoreLinuxMuslAmdx64
2024-11-09 16:05:21.6009 Info File /etc/issue: Welcome to Alpine Linux 3.20
2024-11-09 16:05:21.6009 Info Running in Docker: Yes (image build: v0.22.891-ls584)
2024-11-09 16:05:21.6017 Info ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads
2024-11-09 16:05:21.6017 Info App config/log directory: /config/Jackett
2024-11-09 16:05:21.6033 Info Using proxy: Disabled
2024-11-09 16:05:21.6033 Info Using FlareSolverr: No
2024-11-09 16:05:21.6043 Info Using HTTP Client: HttpWebClient2
```

### Description

TV categories show zero results during manual search when searching for all (blank query with TV category(s) selected).
Sonarr refuses to add Jackett as an index with the error:

> **Query successful, but no results in the configured categories were returned from your indexer. This may be an issue with the indexer or your indexer category settings.**

I get results if I manually search for a specific TV show but all search (blank) returns zero results.
Blank search for Movies returns results

### Logged Error Messages

```text
2024-11-09 16:17:26.9400 Info Manual search in XSpeeds => Found 0 releases [342ms]
2024-11-09 16:18:04.0025 Info Manual search in XSpeeds for lost => Found 8 releases [367ms]
2024-11-09 16:18:35.2970 Info Manual search in XSpeeds => Found 30 releases [356ms]
```

### Screenshots

![jck1](https://github.com/user-attachments/assets/66281e47-f71e-44d9-bc87-edcef5f49d27)
![jck2](https://github.com/user-attachments/assets/e554eddb-d3e6-4ccd-b0c1-1fdab45d15cf)
![jck3](https://github.com/user-attachments/assets/e099bed4-f3f6-42fc-b041-62fe48af3860)

**Comment by garfield69** [comment]
It is a limitation of the web site which does not support multi-category searching.

So the indexer is set up to accept a single category search, but when you specify multi-categories the indexer uses cat=0 which means ALL to the web site search engine.
Problem is that you only get the first page of results, and if there happens to be just TV on the first page and you were looking for Movies then you are out of luck.

However, there should be  a workaround.
If you want to look for ALL  movies, the WEB site uses category 12.
Unfortunately, while the query sent to the web site contains category=12, the returned results for titles contains the specific category, for example 145 for movies HD, and 146 for movies SD, etc, and Jackett thinks none of these cats are what was asked for (category 12) so it drops them out of the results ;-D
When the query cat=0 Jackett accepts all the categories from the results. 

So its a catch 22.

I suppose the simple solve is the ignore the category that comes with the title and return the cat from the query.
It would mean that keywordless searches for all moves (cat=12) would not show the correct detailed category (movies HD, Movies SD, Movies 4k, Movies HECV), and it would show all of them as just Moves.
But at least that would give results.

So in summary, we could, for keywordless searches with a single category that the web site uses for multiple categories (12 for ALL movies, 11 All Movies Boxsets, 13 All Music, 126 All TV, 21 All TV Boxsets) we ignore the actual returned category and used the cat from the query.

Question is, is that worth the effort?

**Comment by garfield69** [comment]
Tested fix with Radarr and Sonarr, was able to add indexer and search for titles.
Available in the next Jackett release due out in about a couple hours.

**Comment by garfield69** [comment]
Actually, I've reverted this fix, since it does nothing to improve the results.
I realised that for keywordless searches, you are totally reliant on there being at least one of the first category being searched for in the first page of results, and my supposed fix does nothing to alter that fact.
You just need to set up to sonarr or radarr indexer at a time when the web site has some of the categories you have set up in your app indexer. I usually do not use the torznab master cats in the indexers, but specify just the individual custom cats, 
so for sonarr I used 100127, 100128, 100129, 100130 
and for radarr 100100, 100117, 100145, 100146
and I was able to save the app indexers.
